packages/react-router/__tests__/useLocation-test.tsx

```diff
@@ -84,4 +85,21 @@ describe("useLocation", () => {
       </pre>
     `);
   });
+
+  // eslint-disable-next-line jest/expect-expect -- type tests
+  it("returns an object with the correct type", () => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars -- used for type tests
+    function TestUseLocationReturnType() {
+      let location = useLocation();
+
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars -- type test
+      type Test1 = Expect<Equal<typeof location.hash, "" | `#${string}`>>;
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars -- type test
+      type Test2 = Expect<Equal<typeof location.pathname, "" | `/${string}`>>;
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars -- type test
+      type Test3 = Expect<Equal<typeof location.search, "" | `?${string}`>>;
+
+      return null;
+    }
+  });
```

Comment:
This doesn't really do much for testing the types. It's just asserting that the code in the library matches the code here. 

A more useful test would be to use `@ts-expect-error` and assert where the tests should fail or not depending on the format of a string being assigned to the various properties. Basically, we should have something testing behaviors, rather than simple equality.

Reply:
Thanks for the feedback, I gave it another shot